py/serial_keel.py

```diff
@@ -286,12 +286,14 @@ def queued(response):
         self.logger.debug(f'In control of {endpoint}')
         return endpoint
 
-    async def control_any(self, label: Union[List[str], str]):
+    async def control_any(self, label: Union[List[str], str], queue_timeout: float = None):
         """
         Start controlling any endpoint matching the given label.
         """
         await self.skws.control_any(label)
-        response = await asyncio.wait_for(self.responses[MessageType.CONTROL].get(), self.timeout)
+        response = await asyncio.wait_for(
```

Comment:
Could we perhaps wrap this in a `try .. except`?
If the timeout occurs, we could then raise the exception further but add some context:

- `label`
- `timeout`
- which "type" of timeout it was (if it's the one from self or from the queue param)

Reply:
Good idea.